Sources/App/Core/SiteURL.swift

```diff
@@ -138,21 +142,28 @@ enum SiteURL: Resourceable {
             case .package:
                 fatalError("pathComponents must not be called with a value parameter")
 
-            case .images:
+            case .images, .stylesheets:
                 fatalError("invalid resource path for routing - only use in static HTML (DSL)")
         }
     }
 
-    static func relativeURL(for path: String) -> String {
+    static let _relativeURL: (String) -> String = { path in
         guard path.hasPrefix("/") else { return "/" + path }
         return path
     }
 
-    static func absoluteURL(for path: String) -> String {
-        Current.siteURL() + relativeURL(for: path)
+    #if DEBUG
+    // make `var` for debug so we can dependency inject
+    static var relativeURL = _relativeURL
+    #else
+    static let relativeURL = _relativeURL
+    #endif
```

Comment:
This is our only change to support dependency injection in the tests: allow `DEBUG` builds to change the `relativeURL` implementation. We don’t need to make anything env specific anymore.

Reply:
The one caveat is that `relativeURL()` cannot have method parameters when it’s a closure but since it’s a single, obvious parameter I’d say that’s fine?

I’ve made the same change for `absoluteURL` purely for symmetry, even though it’s still a function.